core/src/client/lwm2m_static.c

```diff
@@ -318,6 +318,93 @@ int AwaStaticClient_Process(AwaStaticClient * client)
     return result;
 }
 
+static AwaLwm2mResult AwaStaticClientDefaultHandler(AwaStaticClient * client, AwaOperation operation, AwaObjectID objectID, 
+                                                    AwaObjectInstanceID objectInstanceID, AwaResourceID resourceID, AwaResourceInstanceID resourceInstanceID, 
+                                                    void ** dataPointer, uint16_t * dataSize, bool * changed)
+{
+    AwaLwm2mResult result;
+
+    ObjectDefinition * objectDefinition = Definition_LookupObjectDefinition(Lwm2mCore_GetDefinitions(client->Context), objectID);
+    if (objectDefinition != NULL)
+    {
+        uint8_t * offset;
+        ResourceDefinition * resourceDefinition;
+
+        // check instance range
+        if ((objectInstanceID >= 0) && (objectInstanceID < objectDefinition->MaximumInstances))
+        {
+            switch (operation)
+            {
+            case AwaOperation_CreateObjectInstance:
+                result = AwaLwm2mResult_SuccessCreated;
+                break;
+
+            case AwaOperation_DeleteObjectInstance:
+                result = AwaLwm2mResult_SuccessDeleted;
+                break;
+
+            case AwaOperation_CreateResource:
+                result = AwaLwm2mResult_SuccessCreated;
+                break;
+
+            case AwaOperation_DeleteResource:
+                break;
+
+            case AwaOperation_Write:
+            
+                resourceDefinition = Definition_LookupResourceDefinitionFromObjectDefinition(objectDefinition, resourceID);
+                if (resourceDefinition != NULL)
+                {
+                // TODO: check dataSize vs definition->DataElementSize
+                // TODO: what do we do about storing the length.. for opaque etc...
```

Comment:
We have to store how much of the opaque buffer is used, I don't think there is a way around that. The user of the API is supplying us with the maximum amount of data (perhaps we need to rename dataElementSize to dataMaximumElementSize) that can be stored in each element. We might need a function to set/get how much of the opaque data element has been used for a given resource instance.